src/main/java/dataLoader/jsonDataLoader.java

```diff
@@ -0,0 +1,40 @@
+package dataLoader;
+
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Simple Class that parses data set from a json file
+ * without knowing the files format
+ * & prints out the key value pairs.
+ */
+
+
+
+public class jsonDataLoader {
+
+    private String location = "where the file is located";
+    private File json = new File(location);
+
+    public void parseJsonData(String json) throws IOException {
+
+
+        JsonFactory factory = new JsonFactory();
+
+        ObjectMapper mapper = new ObjectMapper(factory);
+        JsonNode rootNode = mapper.readTree(json);
```

Comment:
```readTree``` always returns a not ```null``` ```JsonNode```? If not we need to check before using it and act somehow if it is ```null```.

Reply:
here I can make an if statement saying sth like, if rootNode != null then do ... else print "no can do sorry..."
is this what you mean?

Reply:
Yes sth like this I had in mind also...Perhaps throw an exception if the rootNode is ```null```...However check first with the ```readTree``` function maybe it throws if it cannot establish the root Node

Reply:
from the JavaDoc 

> If a low-level I/O problem (missing input, network error) occurs, a IOException will be > thrown. If a parsing problem occurs (invalid JSON), JsonParseException will be thrown. If no > content is found from input (end-of-input), Java null will be returned.

Reply:
if we need anything else please let me know.